### PR TITLE
Feat descriptions

### PR DESCRIPTION
**Pull request by zkwolf**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)
🔥 增加了一个新组件 Descriptions。

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

此pr包含一个bug修复，修复了在v-slot场景下使用getComponentFromProp无法获取slot的bug
（另外目前Descriptions缺失了一些测试）

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/_util/props-util.js [hunk 1]
```diff
@@ -129,6 +129,10 @@ const getComponentFromProp = (instance, prop, options = instance, execute = true
     if (temp !== undefined) {
       return typeof temp === 'function' && execute ? temp(h, options) : temp;
     }
+    const slotScope = getScopedSlots(instance)[prop];
+    if (slotScope !== undefined) {
+      return typeof slotScope === 'function' && execute ? slotScope(h, options) : slotScope;
+    }
```
Comment (tangjinzhou) [inline components/_util/props-util.js]: 这个有没有具体场景示例，如果确认有问题，这个问题可以提交到 1.4.0 分支 ？

Comment (zkwolf) [reply components/_util/props-util.js]: 唔，比如select optgroup分组那个demo吧，用v-slot:label改写就有问题，使用slot和v-slot在VNode里的数据构造位置不太一样,于是导致用getComponentFromProp(child, 'xxx')这样的代码在v-slot的时候都取不到slot

**Comment by tangjinzhou** [comment]
是基于 antd 哪个分支去做的  统一基于antd 3.23.6的tag分支同步吧

**Comment by zkwolf** [comment]
@tangjinzhou 我从master同步的，目前组件代码和3.23.6 tag同步，上面那个demo修改是本来写的就有点问题，我和ant design那边问了一下，他们改了这边我也同步了，缺个snapshot更新，还少个index.test.js同步

File: components/descriptions/Col.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const ColProps = {
+  child: PropTypes.any,
+  bordered: PropTypes.bool,
+  colon: PropTypes.bool,
+  type: PropTypes.oneOf(['label', 'content']),
+  layout: PropTypes.oneOf(['horizontal', 'vertical']),
+};
+
+const Col = {
```
Comment (tangjinzhou) [inline components/descriptions/Col.jsx]: 使用函数式组件吧

File: components/descriptions/Col.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    if (layout === 'vertical') {
+      labelProps.attrs.colSpan = span * 2 - 1;
+    }
+
+    if (bordered) {
+      if (type === 'label') {
+        return <th {...labelProps}>{label}</th>;
+      }
+      return (
+        <td class={classnames(`${prefixCls}-item-content`)} key="content" colSpan={span * 2 - 1}>
```
Comment (tangjinzhou) [inline components/descriptions/Col.jsx]: 可以直接 `${prefixCls}-item-content`

Comment (tangjinzhou) [reply components/descriptions/Col.jsx]: 以下多个地方同理

File: components/descriptions/Col.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            <span class={`${prefixCls}-item-content`} key="content">
+              {slots.default}
+            </span>
+          </td>
+        );
+      }
+      return (
+        <td colSpan={span} class={classnames(`${prefixCls}-item`)}>
+          <span
+            class={classnames(`${prefixCls}-item-label`, { [`${prefixCls}-item-colon`]: colon })}
```
Comment (tangjinzhou) [inline components/descriptions/Col.jsx]: classnames 实际上在 vue 中并不是刚需，vue底层本身自带了 classnames 的功能，所以可以直接传递 对象 或 数组

File: components/descriptions/demo/basic.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+#### 基本
+简单的展示。
+</cn>
+
+<us>
+#### Basic
+Simplest Usage.
+</us>
+
+```html
```
Comment (tangjinzhou) [inline components/descriptions/demo/basic.md]: 1.5.0 rebase 到了最新 master，后续 demo 都用 ```tpl 来标示

File: components/descriptions/index.jsx [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    ret = [];
+  } else if (!Array.isArray(value)) {
+    ret = [value];
+  }
+  return ret;
+}
+
+export const DescriptionsItem = {
+  name: 'ADescriptionsItem',
+  props: initDefaultProps(DescriptionsItemProps, { span: 1 }),
```
Comment (tangjinzhou) [inline components/descriptions/index.jsx]: 使用函数式组件 and 这里不需要 render ？

Comment (zkwolf) [reply components/descriptions/index.jsx]: 它还真不用render...唯一的作用就是接收prop和在col里被当作child prop传入然后取它的prop来renderCol

File: components/descriptions/demo/basic.md (hunk 5 above)
Comment (zkwolf) [reply components/descriptions/demo/basic.md]: fixed

File: components/descriptions/Col.jsx (hunk 4 above)
Comment (zkwolf) [reply components/descriptions/Col.jsx]: fixed

File: components/descriptions/Col.jsx (hunk 3 above)
Comment (zkwolf) [reply components/descriptions/Col.jsx]: fixed

**Comment by undoZen** [comment]
请问这个什么时候发布？或者能先发一个基于 1.4.8 的 next 版本吗？

**Pull request merged**